### PR TITLE
docs: Clean up SDL docs

### PR DESCRIPTION
**Pull request by vpetrovykh**
Rather than duplicating the synopsys description refer to DDL for most
things and make a separate mention of the nested sub-declarations (liked
to SDL declarations).

### CODE REVIEW

File: docs/edgeql/ddl/index.rst [hunk 1]
```diff
@@ -1,7 +1,10 @@
 .. _ref_eql_ddl:
 
-Data Definition
-===============
+Data Definition (DDL)
+=====================
+
+:edb-alt-title: Data Definition
```
Comment (1st1) [inline docs/edgeql/ddl/index.rst]: Data Definition Language (DDL)

Comment (1st1) [reply docs/edgeql/ddl/index.rst]: Please do the same for the SDL section

File: docs/edgeql/sdl/constraints.rst [hunk 2]
```diff
@@ -56,55 +56,8 @@ commands <ref_eql_ddl_constraints>`.
...
-    An optional string literal defining the error message template that
-    is raised when the constraint is violated.  The template is a formatted
-    string that may refer to constraint context variables in curly braces.
-    The template may refer to the following:
-
-    - ``$argname`` -- the value of the specified constraint argument
-    - ``__subject__`` -- the value of the ``title`` attribute of the scalar
-      type, property or link on which the constraint is defined.
+Much of the syntax is the same as for :ref:`DDL <ref_eql_ddl_constraints>`.
+The main difference is in the nested declarations:
```
Comment (1st1) [inline docs/edgeql/sdl/constraints.rst]: I'm not sure I understand what happened here: a bunch of text is deleted and there's nothing below the colon now...

File: docs/edgeql/sdl/index.rst [hunk 3]
```diff
@@ -1,7 +1,10 @@
 .. _ref_eql_sdl:
 
-Schema Definition
-=================
+Schema Definition (SDL)
+=======================
+
+:edb-alt-title: Schema Definition
```
Comment (1st1) [inline docs/edgeql/sdl/index.rst]: `:edb-alt-title: Schema Definition Language (SDL)`

File: docs/edgeql/sdl/links.rst [hunk 4]
```diff
@@ -47,10 +47,42 @@ commands <ref_eql_ddl_links>`.
...
+        [ <index-declarations> ]
+        [ <attribute-declarations> ]
+        ...
+      "}" ]
+
+Description
+-----------
+
+Much of the syntax is the same as for :ref:`DDL <ref_eql_ddl_links>`.
+The main difference is in the nested declarations:
```
Comment (1st1) [inline docs/edgeql/sdl/links.rst]: I guess we better add an example in places like this one. "Much of the syntax is the same" doesn't cut it.

File: docs/edgeql/ddl/index.rst (hunk 1 above)
Comment (vpetrovykh) [reply docs/edgeql/ddl/index.rst]: Title updated

File: docs/edgeql/sdl/index.rst (hunk 3 above)
Comment (vpetrovykh) [reply docs/edgeql/sdl/index.rst]: Title updated

File: docs/edgeql/sdl/links.rst (hunk 4 above)
Comment (vpetrovykh) [reply docs/edgeql/sdl/links.rst]: The example *and* the synopsis already appear earlier in that page. The description of what the synopsis means is long and identical to DDL, hence the link.

File: docs/edgeql/sdl/constraints.rst (hunk 2 above)
Comment (vpetrovykh) [reply docs/edgeql/sdl/constraints.rst]: I removed the description copy-pasted from DDL. The only thing left is a link to it and a valid sub-declaration (with the link to the corresponding SDL). This is now consistent with other SDL sections.

**Comment by vpetrovykh** [comment]
I've made a fair bit of additional changes making sure that the DDL synopsis and descriptions were more explicit, up-to-date and exhaustive since they are the final source of truth.

File: docs/edgeql/admin/roles.rst [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+
+:eql:synopsis:`SET ATTRIBUTE <attribute> := <value>;`
+    Set role *attribute* to *value*.
+    See :eql:stmt:`SET ATTRIBUTE` for details.
+
+
+Examples
+--------
+
+Create a new database:
```
Comment (1st1) [inline docs/edgeql/admin/roles.rst]: new database -> new role

File: docs/edgeql/admin/roles.rst [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+    See :eql:stmt:`DROP ATTRIBUTE <DROP ATTRIBUTE>` for details.
+
+All the subcommands allowed in the ``CREATE ROLE`` block are also
+valid subcommands for ``ALTER ROLE`` block.
+
+
+Examples
+--------
+
+Create a new database:
```
Comment (1st1) [inline docs/edgeql/admin/roles.rst]: Alter role?

File: docs/edgeql/ddl/attributes.rst [hunk 7]
```diff
@@ -38,13 +41,13 @@ has an attribute defined on it, the descendants of that schema item will
...
+The following subcommands are allowed in the
+``CREATE ABSTRACT ATTRIBUTE`` block:
 
-    The following subdefinitions are allowed in the
-    ``CREATE ABSTRACT ATTRIBUTE`` block:
-
-    * :eql:stmt:`SET ATTRIBUTE`
+:eql:synopsis:`SET ATTRIBUTE <attribute> := <value>`
+    Attributes can also have attributes. Set the *attribute* of the
+    enclosing attribute to a specific *value*. See :eql:stmt:`SET
```
Comment (1st1) [inline docs/edgeql/ddl/attributes.rst]: Please make sure `:eql:stmt:` fits on one line -- it's easier to read that way

Comment (vpetrovykh) [reply docs/edgeql/ddl/attributes.rst]: Cleaned those up.

File: docs/edgeql/admin/roles.rst (hunk 6 above)
Comment (vpetrovykh) [reply docs/edgeql/admin/roles.rst]: Fixed

File: docs/edgeql/admin/roles.rst (hunk 5 above)
Comment (vpetrovykh) [reply docs/edgeql/admin/roles.rst]: Fixed

File: docs/edgeql/ddl/links.rst [hunk 8]
```diff
@@ -201,87 +233,73 @@ Parameters
...
-        a union of types.  Only valid for concrete links.
+:eql:synopsis:`SET SINGLE`
+    Change the maximum cardinality of the link set to *one*.  Only
+    valid for concrete links.
 
-    :eql:synopsis:`SET ATTRIBUTE <attribute> := <value>;`
-        Set link item's *attribute* to *value*.
-        See :eql:stmt:`SET ATTRIBUTE` for details.
+:eql:synopsis:`SET MULTI`
+    Change the maximum cardinality of the link set to *greater then one*.
```
Comment (1st1) [inline docs/edgeql/ddl/links.rst]: then -> than

File: docs/edgeql/ddl/props.rst [hunk 9]
```diff
@@ -174,69 +189,62 @@ Parameters
...
+:eql:synopsis:`SET SINGLE`
+    Change the maximum cardinality of the property set to *one*.  Only
+    valid for concrete properties.
 
-    :eql:synopsis:`ALTER TARGET <typename> [, ...]`
-        Change the target type of the property to the specified type or
-        a union of types.  Only valid for concrete properties.
+:eql:synopsis:`SET MULTI`
+    Change the maximum cardinality of the property set to
+    *greater then one*.  Only valid for concrete properties;
```
Comment (1st1) [inline docs/edgeql/ddl/props.rst]: then -> than

File: docs/edgeql/admin/databases.rst [hunk 10]
```diff
@@ -1,10 +1,10 @@
-.. _ref_eql_ddl_databases:
+.. _ref_admin_databases:
 
 =========
 Databases
 =========
 
-This section describes the DDL commands pertaining to
+This section describes the admin commands pertaining to
```
Comment (1st1) [inline docs/edgeql/admin/databases.rst]: admin -> administrative

File: docs/edgeql/admin/roles.rst [hunk 11]
```diff
@@ -0,0 +1,152 @@
...
+        SET allow_login := true;
+        SET password := 'wonderland';
+    };
+
+
+ALTER ROLE
+==========
+
+:eql-statement:
+:eql-haswith:
```
[the remark this answers is not in the document]
Comment (vpetrovykh) [reply docs/edgeql/admin/roles.rst]: Fixed.

File: docs/edgeql/admin/databases.rst (hunk 10 above)
Comment (vpetrovykh) [reply docs/edgeql/admin/databases.rst]: Fixed

File: docs/edgeql/ddl/links.rst (hunk 8 above)
Comment (vpetrovykh) [reply docs/edgeql/ddl/links.rst]: Fixed

File: docs/edgeql/ddl/props.rst (hunk 9 above)
Comment (vpetrovykh) [reply docs/edgeql/ddl/props.rst]: FIxed

**Pull request merged**